### DISCUSSION TITLE
Is `module.register` scoped to the called file or the rest of the process?

### DISCUSSION THREAD

**Discussion by devjiwonchoi** [nodejs/help#4609] [Q&A] [answered]
### Example

While running a process, there's a case where I need to register a hook to resolve TypeScript.
This process is held at `./foo.js`, and is conditional. This `./foo.js` file will be called multiple times.

```js
// foo.js

if (shouldRegisterHook) {
  const { register } = require('module')
  register('./loader.mjs')
}

// Expect this TypeScript file to be imported correctly.
const typescriptFile = await import('path/to/typescript/file.ts')
// ...
```

### Question

1. When `./foo.js` file is called multiple times, can the hook be registered multiple times?
2. What scope does the hook affect in this case?
    a. Throughout the process.
    b. This file `./foo.js` scope.

**Comment by aduh95** [comment, marked as answer]
Calling `register()` affects the whole thread. A hook can be registered several times but I don't really see a use-case for that.

**Reply by devjiwonchoi** [reply]
Yes, it's the opposite, I'm worried about perf as it's unnecessary.
As you mentioned, as [deregister()](https://github.com/nodejs/node/pull/49159) PR is in stale, is there a reason why? Is it alright for me to pick up?
